be/tool/beck.c

```diff
@@ -593,25 +608,34 @@ int main(int argc, char **argv)
 		printf("Cannot find any segment header generation identifer");
 		return EX_DATAERR;
 	}
+	qinit(&s.s_bnode_q, MAX_SCAN_QUEUED);
+	result = M0_THREAD_INIT(&s.s_thread, struct scanner *,
+				NULL, &scanner_thread, &s, "scannner");
+	if (result != 0)
+		err(EX_CONFIG, "Cannot start scanner thread.");
 	/*
 	 * Skip builder related calls for dry run as we will not be building a
 	 * new segment.
 	 */
 	if (!dry_run) {
 		qinit(&q, MAX_QUEUED);
-		qinit(&s.s_bnode_q, MAX_SCAN_QUEUED);
 		s.s_q = b.b_q = &q;
 		result = builder_init(&b);
 		s.s_seg = b.b_seg;
 		m0_be_engine_tx_size_max(&b.b_dom->bd_engine, &max, NULL);
 		s.s_max_reg_size = max.tc_reg_size;
 		if (result != 0)
 			err(EX_CONFIG, "Cannot initialise builder.");
-		result = M0_THREAD_INIT(&s.s_thread, struct scanner *,
-					NULL, &scanner_thread, &s, "scannner");
-		if (result != 0)
-			err(EX_CONFIG, "Cannot initialise builder.");
+	} else {
+		/**
+		 *  Since we do not have builder variables holding segment data,
+		 *  we use the global variable s_seg for this purpose.
+		 */
+		seg_get(s.s_file, &s_seg);
+		s.s_seg = &s_seg;
+		s.s_max_reg_size = DEFAULT_BE_MAX_TX_REG_SZ;
 	}
+
 	result = scanner_init(&s);
```

Comment:
This is also confusing.
Usually, a struct or object should be initialized first, and then used.
But, some of the field members of `s` are already above.